packages/template-retail-react-app/app/hooks/use-dnt-modal.js

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import PropTypes from 'prop-types'
+import {FormattedMessage, useIntl} from 'react-intl'
+import {
+    Button,
+    CloseButton,
+    Box,
+    useDisclosure,
+    Heading,
+    Stack,
+    Text,
+    Flex
+} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {HideOnDesktop, HideOnMobile} from '@salesforce/retail-react-app/app/components/responsive'
+import {useDNT} from '@salesforce/commerce-sdk-react'
+import {useLocation} from 'react-router-dom'
+
+export const DntModal = ({isOpen, onOpen, onClose, ...props}) => {
+    const {dntStatus, updateDNT} = useDNT()
+    const {formatMessage} = useIntl()
+    const location = useLocation()
+
+    useEffect(() => {
+        if (dntStatus === undefined) onOpen()
+    }, [location])
+
+    const onCloseModal = () => {
+        updateDNT(null)
+        onClose()
+    }
+
+    const buttons = (
+        <>
+            <Button
+                bg="white"
+                color="black"
+                border="1px"
+                _hover={{bg: 'gray.100'}}
+                borderColor="gray.100"
+                boxShadow="md"
+                onClick={() => {
+                    updateDNT(true)
+                    onClose()
+                }}
+                width="100%"
+            >
+                <FormattedMessage defaultMessage="Decline" id="dnt_modal.decline" />
+            </Button>
+            <Button
+                onClick={() => {
+                    updateDNT(false)
+                    onClose()
+                }}
+                boxShadow="md"
+                width="100%"
+            >
+                <FormattedMessage defaultMessage="Accept" id="dnt_modal.accept" />
+            </Button>
+        </>
+    )
+
+    const description = (
+        <Text color={'gray.700'} fontWeight={500} marginTop={7}>
+            <FormattedMessage
+                defaultMessage="This website uses cookies to enhance user experience and to analyze performance and traffic on our website. We also share information about your use of our site with our social media, advertising and analytics partners.  By clicking “Accept,” you agree to our website’s cookie use as described in our Cookie Policy. "
+                id="dnt_modal.description"
+            />
+        </Text>
+    )
+
+    return (
+        isOpen && <Box
```

Comment:
Any reason we are moving away from Chakra Modal?

Reply:
The Chakra UI doesn't allow for the content behind the modal to be interactive. 
> Quoting from the docs: [content behind a modal dialog is inert, meaning that users cannot interact with it](https://v2.chakra-ui.com/docs/components/modal)

The most the Chakra UI can allow for the background content is to be able to scroll. I looked for and tried supported props that could be passed into the Modal that could allow for interaction, but there wasn't any that could fix this issue.